tests/libvmifact/architecture.go

```diff
@@ -19,12 +19,51 @@
 
 package libvmifact
 
-var architecture string
+import "kubevirt.io/kubevirt/pkg/libvmi"
+
+const (
+	// required to start qemu on ARM with UEFI firmware
+	// https://github.com/kubevirt/kubevirt/pull/11366#issuecomment-1970247448
+	armMinimalBootableMemory = "128Mi"
+
+	defaultMinimalBootableMemory = "1Mi"
+)
+
+var (
+	architecture string
+	archAware    architectureAwareMemoryProvider = defaultArchMemoryProvider{}
+)
 
 func RegisterArchitecture(arch string) {
 	architecture = arch
+	if architecture == "arm64" {
+		archAware = arm64MemoryProvider{}
+	} else {
+		archAware = defaultArchMemoryProvider{}
+	}
+}
+
+type architectureAwareMemoryProvider interface {
```

Comment:
This solution could have fit a production code, but here it seems an overkill to me.

I can see a few issues with it:
- It adds a large amount of code and abstraction to solve a problem that is easy to identify as is. It helps nothing except avoiding this repeated condition in the tests, which is negligible IMO.
- You keep the old `architecture` variable in place, without a good enough reason, as if something may not be ok with the change. If we cannot be sure it is safe to use as is, it is not worth changing it.
- The original factories took a decision on the amount of memory to serve based not only on the architecture, but also on the OS used. The name used now, `WithArchAwareMemory` or `withRequestedMemory` is hiding that logic. One could use it on the Fedora factory as well based on the name (but that will not be needed).

Reply:
OK. simplified the PR. Now it only exposes `WithCirrosRequestedMemory` and `WithAlpineRequestedMemory`